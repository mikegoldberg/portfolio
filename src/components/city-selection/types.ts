export interface CitySelectionProps {
  city: string;
  onSubmit: (name: string) => void;
}
