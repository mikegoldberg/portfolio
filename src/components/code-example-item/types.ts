import { MouseEventHandler } from "react";

export interface CodeExampleItemProps {
  title: string;
  poster: string;
  onClick: MouseEventHandler;
  source: string;
}
