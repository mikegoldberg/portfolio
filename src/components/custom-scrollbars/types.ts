import { ReactElement } from "react";

export interface CustomScrollbarsProps {
  onScrollFrame: (position: number) => void;
  children?: ReactElement;
}
