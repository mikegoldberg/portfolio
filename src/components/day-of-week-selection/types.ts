type OnDaySelected = {
  (date: string): void;
};

export interface DayOfWeekSelectionProps {
  dates: Array<string>;
  activeDayOfWeek: string;
  onDaySelected: OnDaySelected;
}
