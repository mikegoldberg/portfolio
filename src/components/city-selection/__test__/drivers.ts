export const HEADING = "city-selection-heading";
