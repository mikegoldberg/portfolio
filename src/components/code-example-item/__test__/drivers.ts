export const POSTER = "code-example-item-poster";
export const ONCLICK = "code-example-item-onclick";
export const TITLE = "code-example-item-title";
export const SOURCE = "code-example-item-source";
