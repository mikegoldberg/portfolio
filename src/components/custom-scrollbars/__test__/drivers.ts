export const CONTENT = "custom-scrollbars-content";
