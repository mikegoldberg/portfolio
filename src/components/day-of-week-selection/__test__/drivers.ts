export const DAY_OF_WEEK = "day-of-week-selection-day-of-week";
