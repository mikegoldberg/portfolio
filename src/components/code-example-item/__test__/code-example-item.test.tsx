import { render } from "../../../test-utils";
import CodeExampleItem from "..";
import * as TEST_DRIVERS from "./drivers";
import { CodeExampleItemProps } from "../types";

const defaultProps: CodeExampleItemProps = {
  title: "test city",
  poster: "test_poster",
  onClick: jest.fn(),
  source: "test_source",
};

describe("CodeExampleItem", () => {
  it("renders correctly", () => {
    const { container } = render(<CodeExampleItem {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it("renders the title", () => {
    const { getByTestId } = render(<CodeExampleItem {...defaultProps} />);
    const element = getByTestId(TEST_DRIVERS.TITLE);
    expect(element).toHaveTextContent(defaultProps.title);
  });

  it("renders the poster", () => {
    const { getByTestId } = render(<CodeExampleItem {...defaultProps} />);
    const element = getByTestId(TEST_DRIVERS.POSTER);
    expect(element).toHaveStyle(
      `background-image: url(${defaultProps.poster})`
    );
  });

  it("renders the source", () => {
    const { getByTestId } = render(<CodeExampleItem {...defaultProps} />);
    const element = getByTestId(TEST_DRIVERS.SOURCE);
    expect(element).toHaveAttribute("href", defaultProps.source);
  });
});
