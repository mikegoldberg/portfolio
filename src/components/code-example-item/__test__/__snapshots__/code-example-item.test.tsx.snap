// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CodeExampleItem renders correctly 1`] = `
<div>
  <div
    class="css-4g6ai3"
  >
    <div
      class="css-okkwv0"
    >
      <div
        class="chakra-linkbox css-isg2pc"
      >
        <div
          class="chakra-aspect-ratio css-1uwuser"
          data-testid="code-example-item-poster"
        >
          <div
            class="css-v2kfba"
          />
        </div>
        <a
          class="chakra-linkbox__overlay css-1hnz6hu"
          data-testid="code-example-item-onclick"
          href="#"
        >
          <p
            class="chakra-text css-5yjrf7"
            data-testid="code-example-item-title"
          >
            test city
          </p>
        </a>
      </div>
      <a
        class="chakra-link css-19uv7gw"
        data-testid="code-example-item-source"
        href="test_source"
        rel="noopener"
        target="_blank"
      >
        <div
          class="css-1dnaxzp"
        >
          <p
            class="chakra-text css-0"
          >
            Source
          </p>
          <svg
            fill="none"
            height="1em"
            stroke="currentColor"
            stroke-linecap="round"
            stroke-linejoin="round"
            stroke-width="2"
            viewBox="0 0 24 24"
            width="1em"
            xmlns="http://www.w3.org/2000/svg"
          >
            <desc />
            <path
              d="M0 0h24v24H0z"
              fill="none"
              stroke="none"
            />
            <path
              d="M11 7h-5a2 2 0 0 0 -2 2v9a2 2 0 0 0 2 2h9a2 2 0 0 0 2 -2v-5"
            />
            <line
              x1="10"
              x2="20"
              y1="14"
              y2="4"
            />
            <polyline
              points="15 4 20 4 20 9"
            />
          </svg>
        </div>
      </a>
    </div>
  </div>
  <span
    hidden=""
    id="__chakra_env"
  />
</div>
`;
